specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2020-06-01-preview/apimdeletedservices.json

```diff
@@ -0,0 +1,128 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ApiManagementClient",
+    "description": "Use these REST APIs for performing operations in Azure API Management deployment.",
+    "version": "2020-06-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ApiManagement/deletedservices": {
+      "get": {
+        "tags": [
+          "DeletedService"
+        ],
+        "operationId": "DeletedServices_ListBySubscription",
+        "description": "Lists all soft-deleted services available for undelete for the given subscription.",
+        "x-ms-examples": {
+          "ApiManagementListRegions": {
+            "$ref": "./examples/ApiManagementDeletedServicesListBySubscription.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "List of all soft-deleted services available for undelete for the given subscription.",
+            "schema": {
+              "$ref": "./definitions.json#/definitions/DeletedServicesCollection"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/locations/{location}/deletedservices/{serviceName}": {
```

Comment:
This is better and allows regional routing.  However, I think you need to think through how this scenario works when the resource group has been deleted as well.  This will result in a 404 in that case and the user will have to figure out that they need to create the resource group again to be able to purge which is a pretty bad experience.  This is why keyVault exposes it as a location based API on the subscription (no RG)